### DISCUSSION TITLE
i need rerender after removeQueries

### DISCUSSION THREAD

**Discussion by soheilhasanjani** [TanStack/query#1525] [Q&A]
hi . at the beginning ، components are rendering width data and react-query cache them . but when i use removeQueries , that component dont rerender . help me

**Comment by tannerlinsley** [comment]
Use resetQueries instead.

**Reply by AlanGreene** [reply]
@tannerlinsley Is there a way to use `removeQueries` and trigger a rerender? You suggested using `resetQueries` but that doesn't seem suitable for my use case.

I'm calling `removeQueries` in response to a websocket event notifying of deletion of an item, so I no longer want to cache it (i.e. call `removeQueries`) and know that attempting to refetch it will fail so `resetQueries` doesn't fit here but I do want the subscriber to be updated so I can render the expected error state.

**Reply by pranay06** [reply]
Is there any update on this issue? I have been struggling with this too.

**Comment by TkDodo** [comment]
> I'm calling removeQueries in response to a websocket event notifying of deletion of an item, so I no longer want to cache it (i.e. call removeQueries) and know that attempting to refetch it will fail so resetQueries doesn't fit here but I do want the subscriber to be updated so I can render the expected error state.

if you remove a query and trigger a re-render, the useQuery would have no data, thus go into loading state and try to fetch it. That is exactly what `resetQueries` is doing.

**Reply by kostia1st** [reply]
>if you remove a query and trigger a re-render, the useQuery would have no data, thus go into loading state and try to fetch it. That is exactly what resetQueries is doing.

Unfortunately that's not the case. At least I don't see this happening. Seems like it does NOT remove the former data, instead it "resets" to pre-loaded state. Which, in case of state rehydrated after SSR - is the state you've got from server. Once again - NOT the empty (removed) state, but rather the state you have rehydrated react-query with during page initialization on the client.

I find it extremely frustrating and making the tool (react-query) very unpredictable and hard to understand, and hence - use.

Unless I'm missing something.

**Reply by TkDodo** [reply]
> Which, in case of state rehydrated after SSR - is the state you've got from server.

if you are using `initialData` then yes. If you are using `Hydrate`, then it should work. Please show a reproduction because it's hard to say without seeing the code.
